src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -1476,12 +1476,17 @@ private void doGenCert(String alias, String sigAlgName, InputStream in, PrintStr
                 reqex = (CertificateExtensions)attr.getAttributeValue();
             }
         }
+
+        byte[] signerSubjectKeyIdExt = ((X509Certificate)signerCert).getExtensionValue(
+                KnownOIDs.SubjectKeyID.value());
+
```

Comment:
How about pass in only the KeyIdentifier into the createV3Extensions method? And you can calculated with
```        X509CertImpl impl;
        if (signerCert instanceof X509CertImpl) {
            impl = (X509CertImpl) signerCert;
        } else {
            impl = new X509CertImpl(signerCert.getEncoded());
        }
        impl.getSubjectKeyId();
```

Reply:
Changed as suggested.

Reply:
Sorry, I should have been more verbose on my suggestion. I was thinking about passing in _only_ the `KeyIdentifier` and not `akey`. After all both of them are for the same purpose and it's clear to consolidate to only one. If the cert has an SKID then use it, otherwise calculate one using `new KeyIdentifier(akey)`. All these are done inside the `doGenCert)()` method. The `createV3Extensions` just add an AKID if the parameter is not null.

Reply:
Updated as suggested.